### DISCUSSION TITLE
When the Rust/C version would be released?

### DISCUSSION THREAD

**Discussion by kamyar1979** [icerpc/icerpc-csharp#3946] [Q&A] [answered]
We are writing some adapters for converting our device socket protocol to communicate the IceRPC server. In the future, we will build our own devices with our custom firmware. For embedded programming, .Net is not popular, although we can do it using .NET core or Mono on linux. We prefer to be able to use some bare metal board and a light-weight firmware without OS. It is possible primarily using C and Rust (no-std). When the Rust version would be ready and will it be usable in no-std? The estimated time of arrival for our firmware is about 1 year.

**Comment by pepone** [comment]
Hi Kamyar,

Rust is the next language we want to add support for in IceRPC but we don't have a timeline for it yet. We'll keep you posted.

Cheers,
Jose

**Comment by kamyar1979** [comment]
1. What about no-std compatibility? Is it in the plan?
2. What about C/C++ version?
3. I can contribute for Rust version! How can I help?

**Reply by ReeceHumphreys** [reply, marked as answer]
Hi Kamyar,

1. Planning and discussions for the Rust language mapping are ongoing, so the following is subject to change. We plan on using [Quinn](https://github.com/quinn-rs/quinn) as the QUIC implementation for IceRPC for Rust. Notably, Quinn does not support no-std; see https://github.com/quinn-rs/quinn/issues/579. As such, the initial release will likely not support no-std. We are happy to look into the possibility of adding support for future releases.
2. We currently have no plans for a C/C++ language mapping. However, we are excited to encourage and foster community language mappings! In the coming months, we plan on having a blog post detailing the process of creating a language mapping.
3. The initial development of IceRPC for Rust will be private while we figure out the ideal architecture and API. Once we have a minimal working API, we will make the repository public for community contributions!
